src/core/lib/gprpp/map.h

```diff
@@ -38,6 +39,13 @@ struct StringLess {
   }
 };
 
+template <typename T>
+struct RefCountedPtrLess {
+  bool operator()(const RefCountedPtr<T>& p1, const RefCountedPtr<T>& p2) {
+    return GPR_ICMP(p1.get(), p2.get());
```

Comment:
We need a binary true or false comparison here. True if lhs < rhs and False of lhs > rhs. GPR_ICMP is a tertiary compare operator that returns -1 if lhs < rhs, 0 if lhs == rhs and 1 if lhs > rhs. This will result in the operator returning true of lhs != rhs.

Please replace with 

return p1.get() < p2.get();

Reply:
https://github.com/grpc/grpc/pull/19223 is adding `RefCountedPtrLess` to compare the content that these pointers point to, instead of comparing the addresses.

Reply:
Thanks for pointing this out, Moiz!  I'll fix it.

Juanli, as per my comments in #19223, I think this is actually the right implementation for `RefCountedPtrLess`.  A comparison based on the content of the object being pointed to should be implemented as a separate, custom comparator.